src/background/adblocker.js

```diff
@@ -495,8 +497,22 @@ if (__PLATFORM__ === 'firefox') {
         const { redirect, match } = engine.match(request);
 
         if (redirect !== undefined) {
-          request.blocked = true;
-          result = { redirectUrl: redirect.dataUrl };
+          // There's a possibility that redirecting to file URL can expose
+          // extension existence.
+          if (
+            details.type !== 'xmlhttprequest' &&
+            WEB_ACCESSIBLE_RESOURCES.has(redirect.name)
+          ) {
+            request.blocked = true;
```

Comment:
`request.blocked` can be outside of the if

Reply:
fixed in 9b9ce0badd6bc9d3ab6c06fcf8959ca44c984469